WindowManager/app/src/main/java/com/example/windowmanagersample/SplitLayoutActivity.kt

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.windowmanagersample
+
+import android.graphics.Rect
+import android.os.Bundle
+import android.view.View
+import android.widget.FrameLayout
+import androidx.core.util.Consumer
+import androidx.core.view.doOnLayout
+import androidx.window.DisplayFeature
+import androidx.window.WindowLayoutInfo
+import androidx.window.WindowManager
+import com.example.windowmanagersample.databinding.ActivitySplitLayoutBinding
+
+/** Demo of an Activity splitting views across a [DeviceFeature]. */
+class SplitLayoutActivity : BaseSampleActivity() {
+
+    private lateinit var windowManager: WindowManager
+    private val layoutStateChangeCallback = LayoutStateChangeCallback()
+
+    private lateinit var binding: ActivitySplitLayoutBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySplitLayoutBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+        windowManager = WindowManager(this, getTestBackend())
+        binding.root.doOnLayout {
+            updateWindowLayout(windowManager.windowLayoutInfo)
+        }
+    }
+
+    private fun updateWindowLayout(windowLayoutInfo: WindowLayoutInfo) {
+        val splitPositions = splitViewPositions(
+            binding.contentLayout.root,
+            binding.controlLayout.root,
+            windowLayoutInfo
+        )
+
+        if (splitPositions != null) {
+            val startPosition = splitPositions[0]
+            val startWidthSpec = View.MeasureSpec.makeMeasureSpec(
+                startPosition.width(),
+                View.MeasureSpec.EXACTLY
+            )
+            val startHeightSpec = View.MeasureSpec.makeMeasureSpec(
+                startPosition.height(),
+                View.MeasureSpec.EXACTLY
+            )
+            binding.contentLayout.root.measure(startWidthSpec, startHeightSpec)
```

Comment:
Measuring and laying out a view which is attached to the window, and has a parent is a big foot gun. The parent can overwrite this at any time.

Looks to me like you should write a custom ViewGroup.